dowhy/utils/cit.py

```diff
@@ -105,3 +107,42 @@ def partial_corr(data=None, x=None, y=None, z=None, method="pearson"):
         'p-val': pval.round(5),
     }
     return stats
+
+def entropy(x):
+    """"
+    Returns entropy for a random variable x
+    H(x) = - Σ p(x)log(p(x))
+    """
+    d=defaultdict(lambda:0)
+    s = 0.0
+    entr = 0.0
+    for i in x:
+        d[i]+=1  # Calculating frequency of an event
+        s+=1
+    for i in d:
+        p = d[i]/s # Calculating probability for an event
+        entr-=p*log(p,2) # H(x) = - Σ p(x)log(p(x))
+    return entr
+
+
+def conditional_MI(data=None,x=None,y=None,z=None):
+    """
+    Method to return conditional mutual information between X and Y given Z
+    I(X, Y | Z) = H(X|Z) - H(X|Y,Z)
+                = H(X,Z) - H(Z) - H(X,Y,Z) + H(Y,Z)
+                = H(X,Z) + H(Y,Z) - H(X,Y,Z) - H(Z)
+    :param data : dataset 
+    :param x,y,z : column names from dataset
+    """
+    X = data[x]
+    Y = data[y]
+    t = list(z)
+    Z = data[t]
+    Z = Z.values.tolist()
+    Z = list(data[t].itertuples(index=False, name=None)) 
```

Comment:
why not directly compute conditional entropy H(X|Z) using the formula [here](https://www.sciencedirect.com/topics/computer-science/conditional-entropy#:~:text=The%20conditional%20entropy%20of%20Y%20given%20X%20is,known%20to%20both%20the%20encoder%20and%20the%20decoder.)?  \sum p(x, z) log_2 (x|z)

This way you are estimating four things, each one's error can propagate. In my proposal, you will be estimating only two things. What do you think?